core/xml.js

```diff
@@ -349,6 +349,14 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       }
     } else if (name == 'shadow') {
       goog.asserts.fail('Shadow block cannot be a top-level block.');
+    }  else if (name == 'variables') {
+      if (i == 1) {
```

Comment:
Cool! Yup I figured something like this. I ran into it because I was constructing XML from the scratch vm by doing series of `+= ...` for the tags (no whitespace) and was hitting an error claiming that the variables section wasn't first, though it most definitely was.